test/extended/jenkins/jenkins_plugin.go

```diff
@@ -0,0 +1,160 @@
+	req, err := http.NewRequest(method, uri, body)
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	if body != nil {
+		req.Header.Set("Content-Type", "application/xml")
+		// jenkins will return 417 if we have an expect hdr
+		req.Header.Del("Expect")
+	}
+	req.SetBasicAuth("admin", "password")
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	defer resp.Body.Close()
+
+	o.Expect(err).NotTo(o.HaveOccurred())
+	o.Expect(resp.StatusCode).To(o.BeEquivalentTo(status))
+
+}
+
+func waitForJenkinsActivity(uri, verificationString string, status int) error {
+	consoleLogs := ""
+
+	err := wait.Poll(1*time.Second, 3*time.Minute, func() (bool, error) {
+		req, err := http.NewRequest("GET", uri, nil)
+		if err != nil {
+			return false, err
+		}
+		req.SetBasicAuth("admin", "password")
+		client := &http.Client{}
+		resp, err := client.Do(req)
+		defer resp.Body.Close()
+		if err != nil {
+			return false, err
+		}
+		if resp.StatusCode == status {
+			if len(verificationString) > 0 {
+				contents, err := ioutil.ReadAll(resp.Body)
+				if err != nil {
+					return false, err
+				}
+				consoleLogs = string(contents)
+				if strings.Contains(consoleLogs, verificationString) {
+					return true, nil
+				} else {
+					return false, nil
+				}
+			} else {
+				return true, nil
+			}
+		}
+		return false, nil
+	})
+
+	if err != nil {
+		return fmt.Errorf("got error %v waiting on uri %s with verificationString %s and last set of console logs %s", err, uri, verificationString, consoleLogs)
+	} else {
+		return nil
+	}
+}
+
+// as g.It's are added conceivably will be with new jenkins job def's
+func jenkinsJobBytes(filename, namespace string) []byte {
+	pre := exutil.FixturePath("fixtures", filename)
+	post := exutil.ArtifactPath(filename)
+	err := exutil.VarSubOnFile(pre, post, "PROJECT_NAME", namespace)
+	o.Expect(err).NotTo(o.HaveOccurred())
+	data, err := ioutil.ReadFile(post)
+	o.Expect(err).NotTo(o.HaveOccurred())
+	return data
+}
+
+var _ = g.Describe("jenkins: plugin: run job leverging openshift pipeline plugin", func() {
+	defer g.GinkgoRecover()
+	var oc = exutil.NewCLI("jenkins-plugin", exutil.KubeConfigPath())
+	var hostPort string
+
+	g.BeforeEach(func() {
+
+		g.By("set up policy for jenkins jobs")
+		err := oc.Run("policy").Args("add-role-to-user", "edit", "system:serviceaccount:"+oc.Namespace()+":default").Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("kick off the build for the jenkins ephermeral and application templates")
+		jenkinsEphemeralPath := exutil.FixturePath("fixtures", "jenkins-ephemeral-template.json")
+		err = exutil.NewApp(jenkinsEphemeralPath, oc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		jenkinsApplicationPath := exutil.FixturePath("fixtures", "application-template.json")
+		err = exutil.NewApp(jenkinsApplicationPath, oc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("tag the app image stream")
+		err = exutil.TagImageStream("nodejs-010-centos7", "openshift/nodejs-010-centos7:latest", oc)
```

Comment:
i worked without the `oc tag` ... i'll leave the utility method in the utils package in there unless you disagree - seems like it could be useful down the road for somebody